File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/news/NewsListContract.java
package xuk.imaginary.gui.nets.news;

import xuk.imaginary.BasePresenter;
import xuk.imaginary.BaseView;
import xuk.imaginary.data.NetsSummary;

import java.util.List;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface NewsListContract {

    interface View extends BaseView<Presenter> {

        void addList(boolean isRefresh, List<NetsSummary> netsSummaries);
    }

    interface Presenter extends BasePresenter {
        void getList(boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/news/NewsListContract.kt
package xuk.imaginary.gui.nets.news

import xuk.imaginary.BasePresenter
import xuk.imaginary.BaseView
import xuk.imaginary.data.NetsSummary

/**
 * This specifies the contract between the view and the presenter.
 */
interface NewsListContract {

  interface View : BaseView<Presenter> {

    fun addList(isRefresh: Boolean, netsSummaries: List<NetsSummary>)
  }

  interface Presenter : BasePresenter {
    fun getList(isRefresh: Boolean)
  }
}
